http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -251,16 +269,33 @@ class LeoPubsubMessageSubscriber[F[_]: Async: Timer: ContextShift: Concurrent](
     for {
       runtimeOpt <- clusterQuery.getClusterById(msg.runtimeId).transaction
       runtime <- runtimeOpt.fold(
-        Async[F].raiseError[Runtime](PubsubHandleMessageError.ClusterNotFound(msg.runtimeId, msg))
-      )(Async[F].pure)
+        F.raiseError[Runtime](PubsubHandleMessageError.ClusterNotFound(msg.runtimeId, msg))
+      )(F.pure)
       _ <- if (runtime.status != RuntimeStatus.Stopping)
-        Async[F].raiseError[Unit](
+        F.raiseError[Unit](
           PubsubHandleMessageError.ClusterInvalidState(msg.runtimeId, runtime.projectNameString, runtime, msg)
         )
-      else Async[F].unit
+      else F.unit
       runtimeConfig <- RuntimeConfigQueries.getRuntimeConfig(runtime.runtimeConfigId).transaction
-      _ <- runtimeConfig.cloudService.interpreter
-        .stopRuntime(StopRuntimeParams(RuntimeAndRuntimeConfig(runtime, runtimeConfig), now))
+      _ <- if (runtime.welderEnabled) {
+        welderDAO
+          .flushCache(runtime.googleProject, runtime.runtimeName)
+          .handleErrorWith(e => logger.error(e)(s"Failed to flush welder cache for ${runtime.projectNameString}"))
+      } else F.unit
+      _ <- if (runtimeConfig.cloudService == CloudService.GCE)
+        for {
+          operation <- googleComputeService.stopInstance(runtime.googleProject,
```

Comment:
Same as above -- use the interpreter instead of making Google calls directly

Reply:
same as `deleteRuntime`, we need an `Operation` back here for polling

Reply:
I see.. but the interpreter does extra stuff besides `googleComputeService.stopInstance`. It also flushes the welder cache, builds the shutdown script, and sets instance metadata. I don't think we should duplicate that logic here. Is there a way to make the interpreter return the operation? Or should we just not use Operations?

Reply:
ah I see...I was wondering where welder flush cache logic has moved to...right now, I moved flush welder cache to pubsub function...I'll think about it...I'd still like to use Operation if possible since it does simplify the logic a bit for `delete` and `stop`